site.maintenance.find-untagged.yml

```diff
@@ -0,0 +1,58 @@
+- hosts: localhost
+  connection: local
+  vars:
+    _ec2_whitelist:
+      - "i-07c245fc7984e0927"
+    _ec2_filters:
+      - "running"
+      - "stopped"
+    _ec2_ebs_list_review: []
+    _ec2_untagged_list: []
+  roles:
+     - role: cs.aws-rds-facts
+  tasks:
+    - name: List of mageops instances
+      ec2_instance_info:
+        filters:
+          "tag:Infrastructure": 'mageops'
+          "tag:Tool": 'ansible'
+          instance-state-name: "{{ _ec2_filters }}"
+        region: "{{ aws_region }}"
+      register: _ec2_list_mageops
+
+    - name: List of all EC2 instances
+      ec2_instance_info:
+        filters:
+          instance-state-name: "{{ _ec2_filters }}"
+        region: "{{ aws_region }}"
+      register: _ec2_list_all
+
+    - name: Generate list EC2 to review
+      set_fact:
+        _ec2_untagged_list: "{{ _ec2_untagged_list + [{ 'id' : item.instance_id, 'name' : item.tags.Name }] }}"
+      when: item.instance_id not in _ec2_whitelist
+      with_items: "{{ _ec2_list_all.instances | difference(_ec2_list_mageops.instances) }}"
+
+    - name: Find unused EBS volumes
+      ec2_vol_info:
+        filters:
+          status: "available"
+        region: "{{ aws_region }}"
+      register: _ec2_ebs_list
+
+    - name: Set unused EBS volumes
+      set_fact:
+        _ec2_ebs_list_review: "{{ _ec2_ebs_list_review + [item.id] }}"
+      with_items: "{{ _ec2_ebs_list.volumes }}"
+
+    - name: Set _ec2_list_review
+      set_fact:
+        _ec2_list_review: "{{ _ec2_list_review | default() + 'ID : ' + item.id + ' Name : ' + item.name + '\n' }}"
```

Comment:
does that syntax even works?
not sure what is going on here

Reply:
Changed to more readable